OPHD/Technology/TechnologyReader.cpp

```diff
@@ -104,5 +131,32 @@ void TechnologyReader::readTechnology(NAS2D::Xml::XmlElement& technology, const
 
 void TechnologyReader::readEffects(NAS2D::Xml::XmlElement& effects, Technology& technology)
 {
+	for (auto effectElement = effects.firstChildElement(); effectElement; effectElement->nextSiblingElement())
+	{
+		const std::string effectName = effectElement->value();
+		const std::string effectValue = effectElement->getText();
+		auto effectAttributes = NAS2D::attributesToDictionary(*effectElement);
 
+		if (effectName == "modifier")
+		{
+			technology.effects.push_back(std::make_unique<Technology::Effect>(
+				Technology::Modifier(
+					StringToModifier.at(effectAttributes.get<std::string>("type")),
+					std::stof(effectValue))
+				));

```

Comment:
I believe `std::make_unique` is intended to be used with the exact derived type, rather than the base class type. The parameters should be constructor parameters, rather than a fully constructed object.
```cpp
technology.effects.push_back(
  std::make_unique<Technology::Modifier>(
    StringToModifier.at(effectAttributes.get<std::string>("type")),
    std::stof(effectValue))
  )
);
```

Reply:
I think you're right since I had to do this in a sort of round about way. The above discussion about making Modifier and Unlock separate classes will resolve this.